Sources/Controllers/TargetMenu/GPX/TrackMenu/OATrackMenuAppearanceHudViewController.mm

```diff
@@ -422,7 +510,29 @@ - (void)generateData
                     kCellType: [OAIconTitleValueCell getCellIdentifier],
                     kCellTitle: OALocalizedString(@"reset_to_original"),
                     kCellRightIconName: @"ic_custom_reset",
-                    kCellToggle: @YES
+                    kCellToggle: @YES,
+                    kCellButtonPressed: ^() {
+                        [[OAGPXDatabase sharedDb] reloadGPXFile:[_app.gpxPath stringByAppendingPathComponent:self.gpx.gpxFilePath]
+                                                     onComplete:^{
+                            dispatch_async(dispatch_get_main_queue(), ^{
+                                self.gpx = [[OAGPXDatabase sharedDb] getGPXItem:self.gpx.gpxFilePath];
+                                [self updateGpxData];
+                                [self commonInit];
```

Comment:
Never call commonInit outside of init method. If you need to change somenting, create a separate method